### PR TITLE
[CLEANUP] Remove EMBER_METAL_TRACKED_PROPERTIES feature flag.

### PR DESCRIPTION
**Pull request by rwjblue**
This has been enabled by default since Ember 3.13, it is safe to remove the flagging infrastructure now.

### CODE REVIEW

File: packages/@ember/-internals/metal/lib/properties.ts [hunk 1]
```diff
@@ -4,13 +4,11 @@
...
 import { setWithMandatorySetter } from '@ember/-internals/utils';
-import { EMBER_METAL_TRACKED_PROPERTIES } from '@ember/canary-features';
 import { assert } from '@ember/debug';
 import { DEBUG } from '@glimmer/env';
 import { Decorator } from './decorator';
 import { descriptorForProperty, isClassicDecorator } from './descriptor_map';
 import { revalidateObservers } from './observer';
-import { overrideChains } from './property_events';
 
 export type MandatorySetterFunction = ((this: object, value: any) => void) & {
```
Comment (pzuraq) [inline packages/@ember/-internals/metal/lib/properties.ts]: I believe this can be removed too, since the mandatory setter stuff has all been pulled into `utils`

File: packages/@ember/-internals/metal/lib/watch_key.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import { Meta, meta as metaFor, peekMeta } from '@ember/-internals/meta';
```
Comment (pzuraq) [inline packages/@ember/-internals/metal/lib/watch_key.ts]: I believe this entire file can be removed

File: packages/@ember/-internals/metal/lib/decorator.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { Meta, meta as metaFor } from '@ember/-internals/meta';
```
Comment (pzuraq) [inline packages/@ember/-internals/metal/lib/decorator.ts]: I believe `addDependentKeys` from this file can be removed as well

**Review by pzuraq — COMMENTED** [review]
Overall this looks good, we can merge whenever. I'd like to ideally clean up the rest of the code that is now no longer used though, I believe the `chains` file can also be removed entirely after this as well for instance. It'd be good to do this now while it's fresh, so we don't get confused about what is and isn't used anymore.

**Comment by rwjblue** [comment]
@pzuraq - OK, I _think_ I've removed all of the things you mentioned above (and also did a bit of sleuthing work to see what dead code remained, see #18578 for that research). This is ready for another look...

**Review by pzuraq — APPROVED** [review]

**Comment by kiwiupover** [comment]
thanks @rwjblue for all your hard work mate!!

**Comment by rwjblue** [comment]
tldr; Looks like this shaves ~ 2KB off of the production build sizes.

Testing with the following:

```
ember new foo
cd foo
yarn link ember-source
ember build -prod
```

These are the gzipped vendor sizes for before/after:

* Before (as of https://github.com/emberjs/ember.js/commit/a590298a3ac9d3199217ae02c56e0d21e809f2fd)
  * 180.96KB - IE11, last Chrome, last Firefox, last Safari
  * 170.35 - last Chrome, last Firefox, last Safari
* After (as of https://github.com/emberjs/ember.js/commit/b4456779d0f5f5b99da853c4e1f0688ce96cc27c)
  * 178.75 - IE11, last Chrome, last Firefox, last Safari
  * 168.19 - last Chrome, last Firefox, last Safari
